prime-router/src/main/kotlin/transport/SoapTransport.kt

```diff
@@ -179,9 +182,44 @@ class SoapTransport(private val httpClient: HttpClient? = null) : ITransport {
                 ", Exception: ${t.localizedMessage}"
             context.logger.severe(msg)
             context.logger.severe(t.stackTraceToString())
-            actionHistory.setActionType(TaskAction.send_warning)
-            actionHistory.trackActionResult(msg)
-            RetryToken.allItems
+            // do some additional handling of the error here. if we are dealing with a 400 error, we
+            // probably don't want to retry, and we need to stop now
+            // if the error is a 500 we can do a retry, but both should probably throw a pager duty notification
+            when (t) {
+                is ClientRequestException -> {
+                    (t as ClientRequestException).let {
+                        context.logger.severe(
+                            "Received ${it.response.status.value}: ${it.response.status.description} " +
+                                "requesting ${it.response.request.url}. This is not recoverable. Will not retry."
+                        )
+                    }
+                    actionHistory.setActionType(TaskAction.send_error)
+                    actionHistory.trackActionResult(msg)
+                    null
+                }
+                is ServerResponseException -> {
```

Comment:
So, I think a SOAP fault will always be an HTTP 500. (See this part of the SOAP spec that I'm referring to https://www.w3.org/TR/2007/REC-soap12-part0-20070427/). A SOAP fault can either be a client or a server error. So, I would be happier if you looked at the SOAP fault code.